### ISSUE TITLE
NSwag / NjsonSchema assembly version conflict

### ISSUE THREAD

**Issue by frankkoenigstein** [NSwag/NSwag#594]
Added via NuGet latest versions of NSwag and NJsonSchema:
```
NJsonSchema.8.3.6252.27806/
NJsonSchema.CodeGeneration.8.3.6252.27823/
NJsonSchema.CodeGeneration.CSharp.8.3.6252.27823/
NJsonSchema.CodeGeneration.TypeScript.8.3.6252.27823/
NSwag.Annotations.9.2.6249.28920/
NSwag.AssemblyLoader.9.2.6249.28922/
NSwag.CodeGeneration.9.2.6249.28920/
NSwag.CodeGeneration.CSharp.9.2.6249.28921/
NSwag.CodeGeneration.TypeScript.9.2.6249.28921/
NSwag.Commands.9.2.6249.28921/
NSwag.Core.9.2.6249.28920/
NSwag.SwaggerGeneration.9.2.6249.28921/
NSwag.SwaggerGeneration.WebApi.9.2.6249.28921/
```

And got the following exception:

```
Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.IO.FileLoadException: Could not load file or assembly 'NJsonSchema, Version=8.1.6249.15627, Culture=neutral, PublicKeyToken=c2f9c3bdfae56102' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Server stack trace:
   at NSwag.SwaggerGeneration.AssemblyTypeToSwaggerGenerator.NetAssemblyLoader.FromAssemblyTypeAsync(String[] classNames, String settingsData)
   at NSwag.SwaggerGeneration.AssemblyTypeToSwaggerGenerator.NetAssemblyLoader.FromAssemblyType(String[] classNames, String settingsData) in C:\projects\nswag\src\NSwag.AssemblyLoaderCore\SwaggerGenerators\AssemblyTypeToSwaggerGenerator.cs:line 87
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]:
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at NSwag.SwaggerGeneration.AssemblyTypeToSwaggerGenerator.NetAssemblyLoader.FromAssemblyType(String[] classNames, String settingsData)
   at NSwag.SwaggerGeneration.AssemblyTypeToSwaggerGenerator.<>c__DisplayClass2_0.<GenerateAsync>b__0() in C:\projects\nswag\src\NSwag.AssemblyLoaderCore\SwaggerGenerators\AssemblyTypeToSwaggerGenerator.cs:line 64
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NSwag.SwaggerGeneration.AssemblyTypeToSwaggerGenerator.<GenerateAsync>d__2.MoveNext() in C:\projects\nswag\src\NSwag.AssemblyLoaderCore\SwaggerGenerators\AssemblyTypeToSwaggerGenerator.cs:line 64
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at avPro.Swagger.SwaggerGenerator.<HandleConfig>d__5.MoveNext() in C:\git\avPro.API\avPro.Swagger\SwaggerGenerator.cs:line 67
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.WaitAll(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.WaitAll(Task[] tasks, Int32 millisecondsTimeout)
   at System.Threading.Tasks.Task.WaitAll(Task[] tasks)
   at avPro.Swagger.SwaggerGenerator.Generate() in C:\git\avPro.API\avPro.Swagger\SwaggerGenerator.cs:line 41
   at avPro.Swagger.Program.Main(String[] args) in C:\git\avPro.API\avPro.Swagger\Program.cs:line 33

```

Looks like some Nswag library still references NJsonSchema 8.1.6249.15627. Please update the library references.

**Comment by rsuter** [comment]
NSwag v9.2 is compiled against NJS v8.1, if you install this version it should work... 
The question is how we can solve this problem when a newer version of NJS is installed.

**Comment by rsuter** [comment]
The next version of NSwag will be compiled against the latest version of NJS, but we will always have this problem if the NJS library is updated after the NSwag release. So we need to find a solution for this scenario...

**Comment by frankkoenigstein** [comment]
I tried the downgrade already. But when I downgrade to NJS 8.1, NJS 8.3 is not found.
Looks like NSwag packages are compiled against differendt versions of NJS:

```
NJsonSchema.8.1.6249.15627/
NJsonSchema.CodeGeneration.8.1.6249.15650/
NJsonSchema.CodeGeneration.CSharp.8.1.6249.15650/
NJsonSchema.CodeGeneration.TypeScript.8.1.6249.15650/
NSwag.Annotations.9.2.6249.28920/
NSwag.AssemblyLoader.9.2.6249.28922/
NSwag.CodeGeneration.9.2.6249.28920/
NSwag.CodeGeneration.CSharp.9.2.6249.28921/
NSwag.CodeGeneration.TypeScript.9.2.6249.28921/
NSwag.Commands.9.2.6249.28921/
NSwag.Core.9.2.6249.28920/
NSwag.SwaggerGeneration.9.2.6249.28921/
NSwag.SwaggerGeneration.WebApi.9.2.6249.28921/
```

```
Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'NJsonSchema, Version=8.3.6252.27806, Culture=neutral, PublicKeyToken=c2f9c3bdfae56102' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) ---> System.IO.FileLoadException: Could not load file or assembly 'NJsonSchema, Version=8.1.6249.15627, Culture=neutral, PublicKeyToken=c2f9c3bdfae56102' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   --- End of inner exception stack trace ---
   at avPro.Swagger.SwaggerGenerator.HandleConfig(NSwagAssemblyConfig assemblyConfig, String solutionDirectory, String swaggerOutputDirectory, String clientOutputDirectory)
   at avPro.Swagger.SwaggerGenerator.<>c__DisplayClass4_0.<Generate>b__0(NSwagAssemblyConfig x) in C:\git\avPro.API\avPro.Swagger\SwaggerGenerator.cs:line 40
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at avPro.Swagger.SwaggerGenerator.Generate() in C:\git\avPro.API\avPro.Swagger\SwaggerGenerator.cs:line 40
   at avPro.Swagger.Program.Main(String[] args) in C:\git\avPro.API\avPro.Swagger\Program.cs:line 33
```

**Comment by rsuter** [comment]
Just released v9.3, can you try with the latest versions of NSwag an NJS?

**Comment by frankkoenigstein** [comment]
With 9.3 my program runs fine.
Thanks.
